app/api_topologies.go

```diff
@@ -31,6 +139,52 @@ type topologyStats struct {
 	FilteredNodes      int `json:"filtered_nodes"`
 }
 
+type byName []APITopologyDesc
+
+func (a byName) Len() int           { return len(a) }
+func (a byName) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a byName) Less(i, j int) bool { return a[i].Name < a[j].Name }
+
+func (r *registry) add(ts ...APITopologyDesc) {
+	r.Lock()
+	defer r.Unlock()
+	for _, t := range ts {
+		t.URL = apiTopologyURL + t.id
+
+		if t.parent != "" {
+			parent := r.items[t.parent]
+			parent.SubTopologies = append(r.items[t.parent].SubTopologies, t)
+			sort.Sort(byName(parent.SubTopologies))
+			r.items[t.parent] = parent
+		}
+
+		r.items[t.id] = t
+	}
+}
+
+func (r *registry) get(name string) (APITopologyDesc, bool) {
+	r.RLock()
+	defer r.RUnlock()
+	t, ok := r.items[name]
+	return t, ok
+}
+
+func (r *registry) walk(f func(APITopologyDesc)) {
+	r.RLock()
+	defer r.RUnlock()
+	descs := []APITopologyDesc{}
+	for _, desc := range r.items {
+		if desc.parent != "" {
+			continue
+		}
+		descs = append(descs, desc)
+	}
+	sort.Sort(byName(descs))
+	for _, desc := range descs {
+		f(desc)
+	}
```

Comment:
Add puts them in a map, not a list.

I considered having add put them in a list as well, and keeping that sorted, but it wasn't clear to me if the k8s topologies were added multiple times, and if add was supposed to dedupe.